// parse5/dist/index.js
import { Parser } from 'parse5/dist/parser/index.d.ts';
export { defaultTreeAdapter } from 'parse5/dist/tree-adapters/default.d.ts';
export { /** @internal */ Parser } from 'parse5/dist/parser/index.d.ts';
export { serialize, serializeOuter } from 'parse5/dist/serializer/index.d.ts';
export { ERR as ErrorCodes } from 'parse5/dist/common/error-codes.d.ts';
/** @internal */
export * as foreignContent from 'parse5/dist/common/foreign-content.d.ts';
/** @internal */
export * as html from 'parse5/dist/common/html.d.ts';
/** @internal */
export * as Token from 'parse5/dist/common/token.d.ts';
/** @internal */
export { Tokenizer, TokenizerMode } from 'parse5/dist/tokenizer/index.d.ts';
// Shorthands
/**
 * Parses an HTML string.
 *
 * @param html Input HTML string.
 * @param options Parsing options.
 * @returns Document
 *
 * @example
 *
 * ```js
 * const parse5 = require('parse5');
 *
 * const document = parse5.parse('<!DOCTYPE html><html><head></head><body>Hi there!</body></html>');
 *
 * console.log(document.childNodes[1].tagName); //> 'html'
 *```
 */
export function parse(html, options) {
    return Parser.parse(html, options);
}
export function parseFragment(fragmentContext, html, options) {
    if (typeof fragmentContext === 'string') {
        options = html;
        html = fragmentContext;
        fragmentContext = null;
    }
    const parser = Parser.getFragmentParser(fragmentContext, options);
    parser.tokenizer.write(html, true);
    return parser.getFragment();
}
